src/services/configService.test.ts

```diff
@@ -170,121 +151,142 @@ describe("Config Service", () => {
+        expect(() => new ConfigService(sls, {} as any))
+          .toThrowError("Runtime nodejs is not supported.\n" + 
+            "Runtimes supported: nodejs10,nodejs12,python3.6,python3.7,python3.8");
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toBeUndefined();
+      });
+    
+      it("throws error when unsupported nodejs version in defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = "nodejs5.x" as any;
+        expect(() => new ConfigService(sls, {} as any))
+          .toThrowError("Runtime nodejs5.x is not supported.\n" + 
+            "Runtimes supported: nodejs10,nodejs12,python3.6,python3.7,python3.8");
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toBeUndefined();
+      });
+    
+      it("Does not throw an error when valid nodejs version is defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = Runtime.NODE10
+        expect(() => new ConfigService(sls, {} as any)).not.toThrow();
+        const expectedRuntime: FunctionRuntime = {
+          language: SupportedRuntimeLanguage.NODE,
+          version: "10"
+        }
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toEqual(expectedRuntime);
+      });
+    
+      it("Does not throw an error when nodejs version is defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = Runtime.NODE10
+        expect(() => new ConfigService(sls, {} as any)).not.toThrow();
+        const expectedRuntime: FunctionRuntime = {
+          language: SupportedRuntimeLanguage.NODE,
+          version: "10"
+        }
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toEqual(expectedRuntime);
+      });
+    
+      it("Does not throw an error when specific nodejs version is defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = Runtime.NODE10;
+        expect(() => new ConfigService(sls, {} as any)).not.toThrow();
+        const expectedRuntime: FunctionRuntime = {
+          language: SupportedRuntimeLanguage.NODE,
+          version: "10"
+        }
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toEqual(expectedRuntime);
+      });
+    
+      it("throws an error when no nodejs version is defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = undefined;
+        expect(() => new ConfigService(sls, {} as any)).toThrowError("Runtime version not specified in serverless.yml");
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toBeUndefined();
+      });
+    
+      it("does not throw an error with python3.6", () => {
```

Comment:
why test `3.6` explicitly and not just valid python version?

Reply:
added tests for the other versions as well. originally it was just 3.6 supported